sdk/inc/azure/platform/az_platform_x509_credential.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Azure x509 credential definitions.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_PLATFORM_X509_CREDENTIAL_H
+#define _az_PLATFORM_X509_CREDENTIAL_H
+
+#include <azure/core/az_credentials_common.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include <stddef.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Defines the key storage types.
+ */
+typedef enum
+{
+  AZ_CREDENTIALS_X509_KEY_MEMORY = 0,
+  AZ_CREDENTIALS_X509_KEY_SECURITY_MODULE = 1,
+} az_platform_credential_x509_key_type;
+
+/**
+ * @brief x509 credential definition.
+ */
+typedef struct
+{
+  struct
+  {
+    /// Contains x509 certificate
+    az_span cert;
+
+    /// Contains x509 key
+    az_span key;
+
+    /// Specifies the storage type for the key.
+    az_platform_credential_x509_key_type key_type;
```

Comment:
Here too, key type relates to the type of the key itself. 
The concept being considered here is: where is this key coming from?
So maybe `storage_source` or `key_source` would be more correct.

Reply:
Changed to `key_source` thanks for pointing it out!